tests/virtualization/xen/virtmanager_add_devices.pm

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: This test adds some devices to our VMs
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
+
+use base "x11test";
+use xen;
+use strict;
+use testapi;
+use utils;
+use virtmanager 'detect_login_screen';
+
+sub run {
+    my ($self) = @_;
+    select_console 'x11';
+
+    x11_start_program 'virt-manager';
+    assert_screen "virt-manager_connected";
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "VM $guest will get some new devices";
+
+        assert_and_dclick "virt-manager_list-$guest";
+        detect_login_screen();
+
+        assert_and_click 'virt-manager_details';
+        assert_and_click 'virt-manager_add-hardware';
+        assert_and_click 'virt-manager_add-storage';
+        if (check_screen 'virt-manager_add-storage-ide') {
+            assert_and_click 'virt-manager_add-storage-ide';
+            assert_and_click 'virt-manager_add-storage-select-xen';
+        }
+        assert_screen 'virt-manager_add-storage-xen';
+        assert_and_click 'virt-manager_add-hardware-finish';
+
+        assert_and_click 'virt-manager_add-hardware';
+        assert_and_click 'virt-manager_add-network';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
+        type_string '00:16:3e:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
```

Comment:
Won't you get mac duplicate mac address if you are unlucky?

Reply:
It didn't happened yet ... I need part of the address fixed to be able to identify it. Maybe also in the future we can switch from bridged network to isolated one...